nextcord/http.py

```diff
@@ -151,48 +471,171 @@ def __exit__(
+    dispatch: :class:`Callable`
+        The dispatcher to use for rate limit events.
+    """
 
     def __init__(
         self,
         connector: Optional[aiohttp.BaseConnector] = None,
         *,
+        max_global_requests: int = 50,
+        time_offset: float = 0.0,
+        default_auth: Optional[str] = None,
+        assume_unsync_clock: bool = False,  # TODO: Think about renaming this?
         proxy: Optional[str] = None,
         proxy_auth: Optional[aiohttp.BasicAuth] = None,
-        loop: Optional[asyncio.AbstractEventLoop] = None,
-        unsync_clock: bool = True,
         dispatch: Callable,
     ) -> None:
-        self.loop: asyncio.AbstractEventLoop = asyncio.get_event_loop() if loop is None else loop
-        self.connector = connector
         self.__session: aiohttp.ClientSession = MISSING  # filled in static_login
-        self._locks: weakref.WeakValueDictionary = weakref.WeakValueDictionary()
-        self._global_over: asyncio.Event = asyncio.Event()
-        self._global_over.set()
-        self.token: Optional[str] = None
-        self.bot_token: bool = False
-        self.proxy: Optional[str] = proxy
-        self.proxy_auth: Optional[aiohttp.BasicAuth] = proxy_auth
-        self.use_clock: bool = not unsync_clock
-        self._dispatch: Callable = dispatch
+        self._connector = connector
+        self._max_global_requests = max_global_requests
+        self._time_offset = time_offset
+        self._default_auth = None
+        # For consistency with possible future changes to set_default_auth.
+        self.set_default_auth(default_auth)
+        self._ratelimit_use_timestamp = not assume_unsync_clock
+        self._proxy = proxy
+        self._proxy_auth: Optional[aiohttp.BasicAuth] = proxy_auth
+        self._dispatch = dispatch
 
         # to mitigate breaking changes
-        self.user_agent: str = _USER_AGENT
+        self._user_agent: str = _USER_AGENT
+
+        self._buckets: dict[str, RateLimit] = {}
+        """{"Discord bucket name": RateLimit}"""
+        self._global_rate_limits: dict[Optional[str], RateLimit] = {}
+        """{"Auth string": RateLimit}, None for auth-less ratelimit."""
+        self._url_rate_limits: dict[tuple[str, str, Optional[str]], RateLimit] = {}
+        """{("METHOD", "Route.bucket", "auth string"): RateLimit} auth string may be None to indicate auth-less."""
+
+    def _make_global_rate_limit(self, auth: Optional[str], max_per_second: int) -> GlobalRateLimit:
+        _log.debug(
+            "Creating global ratelimit for auth %s with max per second %s.",
+            _get_logging_auth(auth),
+            max_per_second,
+        )
+        rate_limit = GlobalRateLimit(time_offset=self._time_offset)
+        rate_limit.limit = max_per_second
+        rate_limit.remaining = max_per_second
+        rate_limit.reset_after = 1 + self._time_offset
+        rate_limit.bucket = f"Global {_get_logging_auth(auth) if auth else 'Unauthorized'}"
```

Comment:
TBF not fussed on this, feel free to resolve
```suggestion
        log_auth = _get_logging_auth(auth)
        _log.debug(
            "Creating global ratelimit for auth %s with max per second %s.",
            log_auth,
            max_per_second,
        )
        rate_limit = GlobalRateLimit(time_offset=self._time_offset)
        rate_limit.limit = max_per_second
        rate_limit.remaining = max_per_second
        rate_limit.reset_after = 1 + self._time_offset
        rate_limit.bucket = f"Global {log_auth if auth else 'Unauthorized'}"
```